dimod/constrained.py

```diff
@@ -1083,21 +1083,88 @@ def lower_bound(self, v: Variable) -> Bias:
+
+        Returns:
+            The number of biases.
+
+        """
+        if vartype is None:
+            def count(qm: Union[QuadraticModel, BinaryQuadraticModel]) -> int:
+                return qm.num_variables + (0 if linear_only else qm.num_interactions)
+        else:
+            vartype = as_vartype(vartype, extended=True)
+
+            def count(qm: Union[QuadraticModel, BinaryQuadraticModel]) -> int:
+                if isinstance(qm, BinaryQuadraticModel):
+                    if qm.vartype is not vartype:
+                        return 0
+                    return qm.num_variables + (0 if linear_only else qm.num_interactions)
+                else:
+                    num_biases = sum(qm.vartype(v) is vartype for v in qm.variables)
+                    if not linear_only:
+                        num_biases += sum(qm.vartype(u) is vartype or qm.vartype(v) is vartype
+                                          for u, v, _ in qm.iter_quadratic())
+                    return num_biases
+
+        return count(self.objective) + sum(count(const.lhs) for const in self.constraints.values())
+
+    def num_quadratic_variables(self, vartype: Optional[VartypeLike] = None, *,
+                                include_objective: Optional[bool] = None,
+                                ) -> int:
+        """The total number of variables with at least one quadratic interaction
+        across the constraints and objective.
+
+        Args:
+            vartype: If provided, only variables of the given type are counted.
+
+            include_objective: Whether to include variables in the objective
+                in the count. Currently defaults to false.
+
+        Return:
+            The number of variables.
+
+        .. deprecated:: 0.10.14
+
+            In dimod 0.12.0``include_objective`` will default to true.
+
+        """
+        if include_objective is None:
+            warnings.warn(
+                "in dimod 0.12.0 the default value of include_objective will change to true. "
+                "To suppress this warning while keeping the existing behavior, set include_objective=False. "
+                "To get the new behavior, set include_objective=True",
+                DeprecationWarning,
+                stacklevel=2,
+                )
+
+        if vartype is None:
+            def count(qm: Union[QuadraticModel, BinaryQuadraticModel]) -> int:
+                return sum(qm.degree(v) > 0 for v in qm.variables)
+        else:
+            vartype = as_vartype(vartype, extended=True)
+
+            def count(qm: Union[QuadraticModel, BinaryQuadraticModel]) -> int:
+                if isinstance(qm, BinaryQuadraticModel):
+                    return sum(qm.degree(v) > 0 for v in qm.variables) if qm.vartype is vartype else 0
+                else:
+                    return sum(qm.vartype(v) is vartype and qm.degree(v) > 0
+                               for v in qm.variables)
```

Comment:
Can you move the ``for`` up? I first parsed this line as ``return sum(...) > 0``, and was confused why it was returning a boolean